tests/installation/usb_install.pm

```diff
@@ -0,0 +1,111 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: prepare USB for baremetal installation via ISO.
+# On a minimum system that is launched by ipxe and stops
+# at sshd-server-started, dd installation iso to one usb,
+# write ignition and combustion config to the other usb,
+# and launch the installation with the iso usb.
+# Maintainer: Xiaoli Ai(Alice) <xlai@suse.com>, qe-virt@suse.de
+
+package usb_install;
+use base 'y2_installbase';
+use strict;
+use warnings;
+
+use utils;
+use testapi;
+use bmwqemu;
+use ipmi_backend_utils;
+use Utils::Architectures;
+use LWP::Simple 'head';
+use Utils::Backends 'use_ssh_serial_console';
+
+sub run {
+    assert_screen('sshd-server-started', 5);
+    use_ssh_serial_console;
+    assert_script_run("set -o pipefail");
+
+    # Specify the two usb drives to use
+    my @usb = split('\n', script_output("ls /dev/disk/by-id/ -l | grep -i usb | grep -i -v -E \"generic|part\" | sed 's#^.*\\\/##'"));
+    record_info("Disk info on the machine:", script_output("ls /dev/disk/by-id/ -l; fdisk -l"));
+    die "No proper usb devices!" unless (scalar(@usb) == 2);
+    my $medium_usb = "/dev/$usb[0]";
+    my $provision_usb = "/dev/$usb[1]";
+    record_info("Pick usb drive $medium_usb to store ISO and $provision_usb to store ignition and combustion config.");
+
+    # Write ignition config to one usb
+    assert_script_run("echo y | mkfs.ext4 $provision_usb", 60);
+    assert_script_run("e2label $provision_usb ignition");
+    assert_script_run("mkdir  -p /mnt");
+    assert_script_run("mount $provision_usb /mnt");
+    assert_script_run("mkdir -p /mnt/ignition");
+    my $cmd = "curl -L "
+      . data_url("virt_autotest/host_unattended_installation_files/ignition/config.ign")
+      . " -o /mnt/ignition/config.ign";
+    script_retry($cmd, retry => 2, delay => 5, timeout => 60, die => 1);
+    save_screenshot;
+    assert_script_run("cat /mnt/ignition/config.ign");
+    save_screenshot;
+
+    # Write combustion script to the same usb with ignition
+    assert_script_run("mkdir -p /mnt/combustion");
+    $cmd = "curl -L "
+      . data_url("virt_autotest/host_unattended_installation_files/combustion/script")
+      . " -o /mnt/combustion/script";
+    script_retry($cmd, retry => 2, delay => 5, timeout => 60, die => 1);
+    save_screenshot;
+    my $SERIALCONSOLE = get_required_var('SERIALCONSOLE');
+    assert_script_run("sed -i 's/SERIALCONSOLE/$SERIALCONSOLE/g' /mnt/combustion/script");
+    assert_script_run("chmod a+x /mnt/combustion/script");
+    assert_script_run("ls -l /mnt/combustion/script && cat /mnt/combustion/script");
+    save_screenshot;
+
+    # Flush to usb stick
+    assert_script_run("sync");
+    assert_script_run("umount -l /mnt");
+    record_info("Ignition and combustion files are successfully downloaded and written to usb $provision_usb.");
```

Comment:
You'd better add script_output("cat ignition; cat combustion") because you do not output them full. assert_scrip_run does not guarantee full output of a file.

Reply:
Now they are complete in screen. Yes, let's add them in case they become longer in future which won't give complete view.

Reply:
Yes. Adding them is better always. Not only for view but also for reference.

Reply:
done